### ISSUE TITLE
Version 2.3.0 : mauvais centrage de la carte suite à une modification dans [MAPCONFIG]

### ISSUE THREAD

**Issue by mgageo** [PnX-SI/GeoNature#838]
Suite à cette modification, la carte se centre sur Briançon avec un niveau de zoom 18 (qui n'est pas disponible dans OpenTopoMap).
Mes modifications

# Configuration cartographique
[MAPCONFIG]
    BASEMAP = [
        {"name"= "OpenTopoMap", "layer" = "//a.tile.opentopomap.org/{z}/{x}/{y}.png", "attribution" = "© OpenTopoMap"},
        {"name"= "GB satellite", "layer" = "//geobretagne.fr/gwc02/service/tms/1.0.0/satellite@EPSG%3A3857@jpeg/{z}/{x}/{-y}.jpg", "attribution" = "© GéoBretagne"},
        {"name" = "GoogleSatellite", "layer" = "//{s}.google.com/vt/lyrs=s&x={x}&y={y}&z={z}","subdomains" = ["mt0", "mt1", "mt2", "mt3"],"attribution" = "© GoogleMap"}
    ]
    # Attention: les coordonnées sont au format [Y, X] - cf leaflet configuration (https://leafletjs.com/reference-1.4.0.html#latlng-l-latlng)
    CENTER = [48.11891,-1.61349]
    # Zoom par défaut
    ZOOM_LEVEL = 14
    # Zoom à partir duquel on peut pointer un releve
    ZOOM_LEVEL_RELEVE = 15
    # zoom appliqué sur la carte lorsque l'on clique sur une liste
    ZOOM_ON_CLICK = 16

**Comment by jbrieuclp** [comment]
Tu as recompilé l'appli après ces modifs ?
`Geonature update_configuration`
https://geonature.readthedocs.io/fr/latest/admin-manual.html#configuration-generale-de-l-application

**Comment by camillemonchicourt** [comment]
Oui il fait relancer la compilation après chaque modification de la configuration.
Attention la doc à jour a changé d'URL mais pour le moment on n'a pas encore pu fermer celle sur Readthedocs.
La doc : http://docs.geonature.fr/admin-manual.html#configuration-generale-de-l-application

**Comment by mgageo** [comment]
Oui j'ai bien recompilé l'application, je modifie le nom d'un des layers pour être certain que la modification a bien été prise en compte.
Dans le fichier ~/geonature/frontend/src/conf/app.config.ts les valeurs sont bien à jour

 "MAPCONFIG": {
  "BASEMAP": [
   {
    "name": "OpenTopoMap",
    "layer": "//a.tile.opentopomap.org/{z}/{x}/{y}.png",
    "attribution": "\u00a9 OpenTopoMap"
   },
   {
    "name": "GB satellite",
    "layer": "//geobretagne.fr/gwc02/service/tms/1.0.0/satellite@EPSG%3A3857@jpeg/{z}/{x}/{-y}.jpg",
    "attribution": "\u00a9 G\u00e9oBretagne"
   },
   {
    "name": "GoogleSatellite",
    "subdomains": [
     "mt0",
     "mt1",
     "mt2",
     "mt3"
    ],
    "layer": "//{s}.google.com/vt/lyrs=s&x={x}&y={y}&z={z}",
    "attribution": "\u00a9 GoogleMap"
   }
  ],
  "ZOOM_LEVEL": 14,
  "CENTER": [
   48.11891,
   -1.61349
  ],
  "ZOOM_LEVEL_RELEVE": 15,
  "ZOOM_ON_CLICK": 16
 },

**Comment by TheoLechemia** [comment]
Problème résolu ?

**Comment by mgageo** [comment]
Non, mais c'est normal j'ai arrêté mes tests : je n'arrive pas à débugger le chargement de la page.
J'ai l'impression qu'il y a une demande asynchrone qui recentre la carte.
Dans la trace réseau du navigateur, j'ai bien les tuiles du secteur demandé dans le fichier de configuration.
Les requêtes partent en http vers opentopomap qui répond en https 200 via un http 301.
Puis il y a une série de réponses en https 404.

**Comment by TheoLechemia** [comment]
C'est parce qu’il y a une donnée "exemple" aux alentours de Briançon. Le centrage de la carte est fait automatiquement sur l'emprise des observations du résultat de la recherche.

**Comment by mgageo** [comment]
Donc OpenTopoMap n'a pas de tuiles autour de Briancon ???
Pourtant quand je clique sur la synthèse, c'est le fond de carte du fichier de configuration qui s'affiche puis quand je clique sur l'une des trois données l'affichage passe sur Briancon.

**Comment by TheoLechemia** [comment]
> Pourtant quand je clique sur la synthèse, c'est le fond de carte du fichier de configuration qui s'affiche puis quand je clique sur l'une des trois données l'affichage passe sur Briancon.

Oui car la donnée exemple est située à Briançon. Quand on clique sur une donnée, le zoom est effectuée sur celle-ci.

> Donc OpenTopoMap n'a pas de tuiles autour de Briancon ???

Si mais opentopomap n'est pas disponible à tous les niveau de zoom. En dezoomant un peu vous devriez voir le fond de carte réapparaître.
Le niveau de zoom au clic sur une observation n'est actuellement pas paramétrable (TODO). Je renomme le ticket, puisque c'est le soucis ici.

**Comment by mgageo** [comment]
Leaflet permet de définir des zoom mini et max ( cf https://leafletjs.com/reference-1.6.0.html#tilelayer )
Par défaut leaflet bloque au zoom 18.
Il y a un paramètre maxNativeZoom qui peut être configuré.

En l'absence d'action sur la page d'accueil, je trouve préférable de rester sur la page configurée, basculer sur la dernière observation est déroutant.

**Comment by TheoLechemia** [comment]
Un 1er commit pour ne pas zoomer sur les "X" dernières observations au premier chargement de la carte. On reste sur l'emprise mise en paramètre.
TODO: paramétrer le niveau de zoom
